File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/device/Device.java
package com.jdroid.android.firebase.fcm.device;

import com.jdroid.android.utils.AndroidUtils;
import com.jdroid.android.utils.AppUtils;
import com.jdroid.android.utils.DeviceUtils;

public class Device {

	private String deviceGroupId;
	private String registrationToken;

	private String deviceBrandName;
	private String deviceModelName;
	private String deviceOsVersion;
	private String appVersionCode;

	public Device(String registrationToken, String deviceGroupId) {
		this.deviceGroupId = deviceGroupId;
		this.registrationToken = registrationToken;
		deviceBrandName = DeviceUtils.INSTANCE.getDeviceManufacturer();
		deviceModelName = DeviceUtils.INSTANCE.getDeviceModel();
		deviceOsVersion = AndroidUtils.getPlatformVersion();
		appVersionCode = AppUtils.getVersionCode().toString();
	}

	public String getRegistrationToken() {
		return registrationToken;
	}

	public String getDeviceGroupId() {
		return deviceGroupId;
	}

	public String getDeviceBrandName() {
		return deviceBrandName;
	}

	public String getDeviceModelName() {
		return deviceModelName;
	}

	public String getDeviceOsVersion() {
		return deviceOsVersion;
	}

	public String getAppVersionCode() {
		return appVersionCode;
	}
}

File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/device/Device.kt
package com.jdroid.android.firebase.fcm.device

import com.jdroid.android.utils.AndroidUtils
import com.jdroid.android.utils.AppUtils
import com.jdroid.android.utils.DeviceUtils

class Device(val registrationToken: String, val deviceGroupId: String?) {

    val deviceBrandName: String?
    val deviceModelName: String?
    val deviceOsVersion: String
    val appVersionCode: String

    init {
        deviceBrandName = DeviceUtils.getDeviceManufacturer()
        deviceModelName = DeviceUtils.getDeviceModel()
        deviceOsVersion = AndroidUtils.getPlatformVersion()
        appVersionCode = AppUtils.getVersionCode().toString()
    }
}
